tests/arp/test_tagged_arp.py

```diff
@@ -0,0 +1,259 @@
+        if isinstance(result, test.dataplane.PollFailure):
+            test.fail("Expected packet was not received on device %d, port %r.\n%s"
+                    % (device_number, port, result.format()))
+
+    for port_group in portchannel_ports:
+        for port in port_group:
+            result = testutils.dp_poll(test, device_number=device_number, port_number=port,
+                                       timeout=timeout, exp_pkt=pkt)
+            if isinstance(result, test.dataplane.PollSuccess):
+                break
+        else:
+            test.fail("Expected packet was not received on device %d, ports %s.\n"
+                    % (device_number, str(port_group)))
+
+
+def verify_arp_packets(ptfadapter, vlan_ports_list, vlan_port, vlan_id, untagged_pkt, masked_tagged_pkt):
+    untagged_dst_ports = []
+    tagged_dst_ports = []
+    untagged_dst_pc_ports = []
+    tagged_dst_pc_ports = []
+    logger.info("Verify packets from ports " + str(vlan_port["port_index"][0]))
+    for port in vlan_ports_list:
+        if vlan_port["port_index"] == port["port_index"]:
+            # Skip src port
+            continue
+        if port["pvid"] == vlan_id:
+            if len(port["port_index"]) > 1:
+                untagged_dst_pc_ports.append(port["port_index"])
+            else:
+                untagged_dst_ports += port["port_index"]
+        elif vlan_id in map(int, port["permit_vlanid"]):
+            if len(port["port_index"]) > 1:
+                tagged_dst_pc_ports.append(port["port_index"])
+            else:
+                tagged_dst_ports += port["port_index"]
+
+    verify_packets_with_portchannel(test=ptfadapter,
+                                    pkt=untagged_pkt,
+                                    ports=untagged_dst_ports,
+                                    portchannel_ports=untagged_dst_pc_ports)
+    verify_packets_with_portchannel(test=ptfadapter,
+                                    pkt=masked_tagged_pkt,
+                                    ports=tagged_dst_ports,
+                                    portchannel_ports=tagged_dst_pc_ports)
+
+
+@pytest.mark.bsl
+def test_tagged_arp_pkt(ptfadapter, vlan_ports_list, duthosts, rand_one_dut_hostname, toggle_all_simulator_ports_to_rand_selected_tor):
+    """
+    Send tagged GARP packets from each port.
+    Verify packets egress without tag from ports whose PVID same with ingress port.
+    Verify packets egress with tag from ports who include VLAN ID but PVID different from ingress port.
+    verify show arp command on DUT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    for vlan_port in vlan_ports_list:
+        port_index = vlan_port["port_index"][0]
+        # Send GARP packets to switch to populate the arp table with dummy MACs for each port
+        # Totally 10 dummy MACs for each port, send 1 packet for each dummy MAC
+        dummy_macs = ['{}:{:02x}:{:02x}'.format(DUMMY_MAC_PREFIX, port_index, i+1)
```

Comment:
>port_index [](http://example.com/codeflow?start=66&length=10)

We have some SKU with so many ports. Will that be a problem if we want to test arp on those topo?